From 81ef4c2b05ed6464f1590acf4ec607eeb21e78ec Mon Sep 17 00:00:00 2001
From: c457 <android.c357@gmail.com>
Date: Tue, 5 Jan 2016 21:06:56 -0600
Subject: [PATCH 20/57] av: squashed support for pre-kitkat audio blobs pre
 kitkat audio legacy policy fix for hotword (ok google) pre-kitkat audio
 policy blobs re-activated.

    libnbaio: Don't call get_presentation_position for pre KitKat blobs
    This fixes a crash with the HTC Tegra3 audio blob where
    mStream->get_presentation_position is not NULL but pointing to a unknown
    position in memory.
---
 media/libnbaio/Android.bp                               | 1 +
 media/libnbaio/AudioStreamOutSink.cpp                   | 2 ++
 media/libnbaio/include/media/nbaio/AudioStreamOutSink.h | 2 ++
 3 files changed, 5 insertions(+)

diff --git a/media/libnbaio/Android.bp b/media/libnbaio/Android.bp
index 4220b778e..8ddba7a3a 100644
--- a/media/libnbaio/Android.bp
+++ b/media/libnbaio/Android.bp
@@ -67,6 +67,7 @@ cc_library_shared {
     cflags: [
         "-Werror",
         "-Wall",
+        "-DHAVE_PRE_KITKAT_AUDIO_BLOB",
     ],
 
     include_dirs: ["system/media/audio_utils/include"],
diff --git a/media/libnbaio/AudioStreamOutSink.cpp b/media/libnbaio/AudioStreamOutSink.cpp
index 85648994a..ab8110135 100644
--- a/media/libnbaio/AudioStreamOutSink.cpp
+++ b/media/libnbaio/AudioStreamOutSink.cpp
@@ -75,6 +75,7 @@ ssize_t AudioStreamOutSink::write(const void *buffer, size_t count)
     }
 }
 
+#ifndef HAVE_PRE_KITKAT_AUDIO_BLOB
 status_t AudioStreamOutSink::getTimestamp(ExtendedTimestamp &timestamp)
 {
     uint64_t position64;
@@ -86,5 +87,6 @@ status_t AudioStreamOutSink::getTimestamp(ExtendedTimestamp &timestamp)
     timestamp.mTimeNs[ExtendedTimestamp::LOCATION_KERNEL] = audio_utils_ns_from_timespec(&time);
     return OK;
 }
+#endif
 
 }   // namespace android
diff --git a/media/libnbaio/include/media/nbaio/AudioStreamOutSink.h b/media/libnbaio/include/media/nbaio/AudioStreamOutSink.h
index 348b4f827..a23540e46 100644
--- a/media/libnbaio/include/media/nbaio/AudioStreamOutSink.h
+++ b/media/libnbaio/include/media/nbaio/AudioStreamOutSink.h
@@ -48,7 +48,9 @@ public:
 
     virtual ssize_t write(const void *buffer, size_t count);
 
+#ifndef HAVE_PRE_KITKAT_AUDIO_BLOB
     virtual status_t getTimestamp(ExtendedTimestamp &timestamp);
+#endif
 
     // NBAIO_Sink end
 
-- 
2.11.0

